saleor/payment/utils.py

```diff
@@ -0,0 +1,210 @@
+    with transaction.atomic():
+        txn, error = gateway.charge(
+            payment, payment_token, **gateway_params)
+        if txn.is_success:
+            payment.charge_status = ChargeStatus.CHARGED
+            payment.captured_amount += txn.amount
+            payment.save(update_fields=['charge_status', 'captured_amount'])
+            order = payment.order
+            if order and order.is_fully_paid():
+                handle_fully_paid_order(order)
+    if not txn.is_success:
+        raise PaymentError(error)
+    return txn
+
+
+@validate_payment
+def gateway_authorize(payment: Payment, payment_token: str) -> Transaction:
+    """Authorizes the payment and creates relevant transaction.
+
+    Args:
+     - payment_token: One-time-use reference to payment information.
+    """
+    clean_authorize(payment)
+
+    gateway, gateway_params = get_payment_gateway(payment.gateway)
+    with transaction.atomic():
+        txn, error = gateway.authorize(
+            payment, payment_token, **gateway_params)
+    if not txn.is_success:
+        raise PaymentError(error)
+    return txn
+
+
+@validate_payment
+def gateway_capture(payment: Payment, amount: Decimal) -> Transaction:
+    clean_capture(payment, amount)
+
+    gateway, gateway_params = get_payment_gateway(payment.gateway)
+    with transaction.atomic():
+        txn, error = gateway.capture(
+            payment, amount=amount, **gateway_params)
+        if txn.is_success:
+            payment.charge_status = ChargeStatus.CHARGED
+            payment.captured_amount += txn.amount
+            payment.save(update_fields=['charge_status', 'captured_amount'])
+            order = payment.order
+            if order and order.is_fully_paid():
+                handle_fully_paid_order(order)
+    if not txn.is_success:
+        raise PaymentError(error)
+    return txn
+
+
+@validate_payment
+def gateway_void(payment) -> Transaction:
+    if not payment.can_void():
+        raise PaymentError('Only pre-authorized transactions can be voided.')
+    gateway, gateway_params = get_payment_gateway(payment.gateway)
+    with transaction.atomic():
+        txn, error = gateway.void(payment, **gateway_params)
+        if txn.is_success:
+            payment.is_active = False
+            payment.save(update_fields=['is_active'])
+    if not txn.is_success:
+        raise PaymentError(error)
+    return txn
+
+
+@validate_payment
+def gateway_refund(payment, amount: Decimal) -> Transaction:
+    if amount <= 0:
+        raise PaymentError('Amount should be a positive number.')
+    if amount > payment.captured_amount:
+        raise PaymentError('Cannot refund more than captured')
+    if not payment.can_refund():
+        raise PaymentError('This payment cannot be refunded.')
+
+    gateway, gateway_params = get_payment_gateway(payment.gateway)
+    with transaction.atomic():
+        txn, error = gateway.refund(payment, amount, **gateway_params)
```

Comment:
I think that would be better to pass the parameters as keywords: `gateway.refund(payment=payment, amount=amount, **gateway_params)`. (idem for the other `gateway.*`)